src/CMakeLists.txt

```diff
@@ -197,9 +197,15 @@ if(SSE2_FOUND OR TARGET_ARCHITECTURE STREQUAL "native")
   endif()
   set(ALICEVISION_HAVE_SSE 1)
 endif()
-if(UNIX AND NOT ALICEVISION_BUILD_COVERAGE)
-  set(CMAKE_C_FLAGS_RELEASE "-O3")
-  set(CMAKE_CXX_FLAGS_RELEASE "-O3")
+
+if(UNIX)
+  if (ALICEVISION_BUILD_COVERAGE)
+    set(CMAKE_C_FLAGS_RELEASE "-O1")
+    set(CMAKE_CXX_FLAGS_RELEASE "-O1")
+  else ()
+    set(CMAKE_C_FLAGS_RELEASE "-O3")
+    set(CMAKE_CXX_FLAGS_RELEASE "-O3")
```

Comment:
I think it's better to set the build as Debug rather than mess with the c flags directly
```
set(CMAKE_BUILD_TYPE "Debug")
```
and optionally warn the user if it was Release so it is clear what's happening.

Even in `CodeCoverage.cmake` they says so
```
if(NOT (CMAKE_BUILD_TYPE STREQUAL "Debug" OR GENERATOR_IS_MULTI_CONFIG))
    message(WARNING "Code coverage results with an optimised (non-Debug) build may be misleading")
```